### PR TITLE
Disable table action links if no samples are seleceted

### PR DESCRIPTION
**Pull request by joshsadam**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

Currently, on the project samples page, the workflow execution and transfer samples links function without any samples being selected.  This PR add a check when the buttons are connected to the dom to see if there are any samples selected for that project, and when the selections are updated the buttons will update.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

**No samples selected**
image

**Samples selected**
image

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

**EDIT**
This also works for buttons and both default and primary CSS button classes)
image

Go to any project samples page (that has samples).  

- If no samples are previously selected on the initial load, the two buttons should be disabled.  
- Select a sample and the buttons should become enabled.
- De-select the sample and again the buttons should be disabled. (try clicking to make sure nothing happens).
- Select samples and refresh the page, the buttons should be enabled.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [X] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/javascript/controllers/selection_controller.js [hunk 1]
```diff
@@ -49,4 +59,10 @@ export default class extends Controller {
       JSON.stringify([...storageValue])
     );
   }
+
+  #updateActinLinks(count) {
```
Comment (deepsidhu85) [inline app/javascript/controllers/selection_controller.js:63]: Can you add a comment to why the `#` is infront of the function name here. Also there's a typo in the function name

File: app/javascript/controllers/action_link_controller.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { Controller } from "@hotwired/stimulus";
+
+export default class extends Controller {
+  static values = { required: { type: Number, default: 0 } };
+
+  #classes = [
```
Comment (deepsidhu85) [inline app/javascript/controllers/action_link_controller.js:6]: Please add a comment as to why the `#` is infront of the variable

File: app/views/projects/samples/_table.html.erb [hunk 3]
```diff
@@ -2,6 +2,7 @@
   id="samples-table"
   class="min-w-full table-fixed dark:divide-slate-600"
   data-controller="selection"
+  data-selection-action-link-outlet=".link"
```
Comment (deepsidhu85) [inline app/views/projects/samples/_table.html.erb:5]: Can we update this to `.action-link` so it's more specific to what its being used for (example here..action)

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This is looking good just a few comments below

File: app/javascript/controllers/selection_controller.js (hunk 1 above)
Comment (joshsadam) [reply app/javascript/controllers/selection_controller.js]: Added in 71d0b560f7ff2efa4c9fa50b28ecf755469fe42a

File: app/javascript/controllers/action_link_controller.js (hunk 2 above)
Comment (joshsadam) [reply app/javascript/controllers/action_link_controller.js:8]: Added in 71d0b560f7ff2efa4c9fa50b28ecf755469fe42a

File: app/views/projects/samples/_table.html.erb (hunk 3 above)
Comment (joshsadam) [reply app/views/projects/samples/_table.html.erb]: Updated in 6777432a27326c3d09e2af26cee263544ca2bc18

File: app/javascript/controllers/action_link_controller.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { Controller } from "@hotwired/stimulus";
+
+export default class extends Controller {
+  static values = { required: { type: Number, default: 0 } };
+
+  // # indicated private attribute or method
```
Comment (deepsidhu85) [inline app/javascript/controllers/action_link_controller.js:6]: Change `indicated` to `indicates` here and in the selection_controller comment

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This is looking good and working as expected. Just one last comment

File: app/javascript/controllers/action_link_controller.js (hunk 4 above)
Comment (joshsadam) [reply app/javascript/controllers/action_link_controller.js]: Updated in c8a0cc132ac6188cc6cb2eb2d6a5da02cb34b6ee

**Review by ksierks — APPROVED** [review]
Thanks for adding this.

**Review by deepsidhu85 — APPROVED** [review]

**Pull request merged**